libs/core/src/lib/menu/directives/menu-addon.directive.ts

```diff
@@ -40,4 +41,21 @@ export class MenuAddonDirective {
     /** @hidden */
     @HostBinding('class.fd-menu__addon-before')
     fdAddonBeforeClass = false;
+
+    /** @hidden */
+    private menuComponent = inject(MenuComponent, { optional: true });
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (this.menuComponent) {
```

Comment:
you can have a method called `setAddons(...)` and change entire block to `this.menuComponent?.setAddons([...this.menuComponent._addons, this])`;